yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -782,7 +785,13 @@ object Client extends Logging {
 
   /**
    * Populate the classpath entry in the given environment map.
-   * This includes the user jar, Spark jar, and any extra application jars.
+   *
+   * User jars are generally not added to the JVM's classpath; those are handled by the AM and
+   * executor backend. When the deprecated `spark.yarn.user.classpath.first` is used, user jars are
```

Comment:
It would be clearer if you explicitly say that `those are loaded by the AM or the executor backend dynamically`, because it's technically added to the JVM's class path, just that it's added later.